twisted/internet/iocpreactor/abstract.py

```diff
@@ -71,8 +72,8 @@ def _dispatchData(self):
                 return False
         remainder = size % self.readBufferSize
         if remainder:
-            self.dataReceived(buffer(self._readBuffers[full_buffers],
-                                     0, remainder))
+            remainderView = memoryview(self._readBuffers[full_buffers])
+            self.dataReceived(remainderView[0:remainder].tobytes())
```

Comment:
calling `tobytes()` on the `memoryview` will copy the underlying array to the new `bytes` object, defeating the purpose of using a `memoryview` in the first place.

if this code used to accept `buffer`, though, it should also accept `memoryview`, since `memoryview`'s API is a superset of `buffer`'s.  so it should be enough to just change this line to:

```python
self.dataReceived(remainderView[0:remainder])
```

(you can remove the 0 too, if that reads better)

here's my citation for the copying claim.  the code for `tobytes()` is [here](https://github.com/python/cpython/blob/2.7/Objects/memoryobject.c#L405-L418); if `PyBuffer_Release` isn't suspicious enough, you can see from the source for [`PyBuffer_ToContiguous`](https://github.com/python/cpython/blob/2.7/Objects/abstract.c#L482-L531) that it [copies](https://github.com/python/cpython/blob/2.7/Objects/abstract.c#L496) the backing array, [even if it's not contiguous](https://github.com/python/cpython/blob/2.7/Objects/abstract.c#L523-L528).